pkg/xray/watches.go

```diff
@@ -0,0 +1,250 @@
+package xray
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"log"
+	"net/http"
+
+	"github.com/go-resty/resty/v2"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type WatchGeneralData struct {
+	Name        *string `json:"name,omitempty"`
+	Description *string `json:"description,omitempty"`
+	Active      *bool   `json:"active,omitempty"`
+}
+
+type WatchFilterValue struct {
+	Key   *string `json:"key,omitempty"`
+	Value *string `json:"value,omitempty"`
+}
+
+type WatchFilter struct {
+	Type  *string `json:"type,omitempty"`
+	Value *string `json:"value,omitempty"`
+}
+
+type WatchProjectResource struct {
+	Type            *string        `json:"type,omitempty"`
+	BinaryManagerId *string        `json:"bin_mgr_id,omitempty"`
+	Filters         *[]WatchFilter `json:"filters,omitempty"`
+	// Watch a repo
+	Name *string `json:"name,omitempty"`
```

Comment:
Again, why all the pointers

Reply:
@chb0github In this case, name is optional. `func PackProjectResources` has a verification if the name is empty: `		if res.Name != nil {
			resourceMap["name"] = res.Name
		}`
If name will be a string, instead of a pointer, you can't compare it with `nil` - Cannot convert 'nil' to type 'string'
Is it right if I'll use this instead: `		if len(res.Name) == 0 {
			resourceMap["name"] = res.Name
		}`
?

Reply:
No need for pointers. Length check is fine.

Also, we should remove `omitempty` from all fields? @chb0github